iot/mqtt_example/cloudiot_mqtt_example_nodejs.js

```diff
@@ -222,88 +352,470 @@ function publishAsync(messagesSent, numMessages) {
           // Note: logging packet send is very verbose
         });
       }
-      // [END iot_mqtt_jwt_refresh]
-      publishAsync(messagesSent + 1, numMessages);
+      publishAsync(
+        mqttTopic,
+        client,
+        iatTime,
+        messagesSent + 1,
+        numMessages,
+        connectionArgs
+      );
     }, schedulePublishDelayMs);
   }, publishDelayMs);
 }
 // [END iot_mqtt_publish]
 
-// [START iot_mqtt_run]
-// The mqttClientId is a unique string that identifies this device. For Google
-// Cloud IoT Core, it must be in the format below.
-const mqttClientId = `projects/${argv.projectId}/locations/${
-  argv.cloudRegion
-}/registries/${argv.registryId}/devices/${argv.deviceId}`;
-
-// With Google Cloud IoT Core, the username field is ignored, however it must be
-// non-empty. The password field is used to transmit a JWT to authorize the
-// device. The "mqtts" protocol causes the library to connect using SSL, which
-// is required for Cloud IoT Core.
-let connectionArgs = {
-  host: argv.mqttBridgeHostname,
-  port: argv.mqttBridgePort,
-  clientId: mqttClientId,
-  username: 'unused',
-  password: createJwt(argv.projectId, argv.privateKeyFile, argv.algorithm),
-  protocol: 'mqtts',
-  secureProtocol: 'TLSv1_2_method',
-};
-
-// Create a client, and connect to the Google MQTT bridge.
-let iatTime = parseInt(Date.now() / 1000);
-let client = mqtt.connect(connectionArgs);
-
-// Subscribe to the /devices/{device-id}/config topic to receive config updates.
-// Config updates are recommended to use QoS 1 (at least once delivery)
-client.subscribe(`/devices/${argv.deviceId}/config`, {qos: 1});
-
-// Subscribe to the /devices/{device-id}/commands/# topic to receive all
-// commands or to the /devices/{device-id}/commands/<subfolder> to just receive
-// messages published to a specific commands folder; we recommend you use
-// QoS 0 (at most once delivery)
-client.subscribe(`/devices/${argv.deviceId}/commands/#`, {qos: 0});
-
-// The MQTT topic that this device will publish data to. The MQTT topic name is
-// required to be in the format below. The topic name must end in 'state' to
-// publish state and 'events' to publish telemetry. Note that this is not the
-// same as the device registry's Cloud Pub/Sub topic.
-const mqttTopic = `/devices/${argv.deviceId}/${argv.messageType}`;
-
-client.on('connect', success => {
-  console.log('connect');
-  if (!success) {
-    console.log('Client not connected...');
-  } else if (!publishChainInProgress) {
-    publishAsync(1, argv.numMessages);
+function mqttDeviceDemo(argv) {
+  // [START iot_mqtt_run]
+  // The mqttClientId is a unique string that identifies this device. For Google
+  // Cloud IoT Core, it must be in the format below.
+  const mqttClientId = `projects/${argv.projectId}/locations/${
```

Comment:
Nit. This formatting looks a little weird to me. In other samples, I have done concatenations for longer strings as:

`averylongstring/withmanypartsthatareverylongpartsthat/logicallycanbegroupedintodifferentetc...`

    let string1 = 'averylongstring';
    let string2 = 'withmanypartsthatareverylongpartsthat';
    let string3 = 'logicallycanbegroupedintodifferentetc...';

`${string1}/${string2}/${string3}`

Reply:
Shortened so linter doesn't complain, but will do in the future.